File: backend/collaboration-service/src/service/question.js
import axios from 'axios';

function getQuestionIdFromDifficulty(difficulty) {
    let questionidMap = {
        EASY: 0,
        MEDIUM: 1,
        HARD: 2,
    };
    return (questionidMap.hasOwnProperty(difficulty))
        ? questionidMap[difficulty]
        : 0;
}

export { getQuestionIdFromDifficulty };

File: questionscraper/app/pydantic_model/Question.py
from enum import Enum
from pydantic import BaseModel

class Difficulty(str, Enum):
    easy = 'easy'
    medium = 'medium'
    hard = 'hard'

class QuestionBase(BaseModel):
    id: int
    href : str
    difficulty : Difficulty
    title : str

class CompanyQuestion(BaseModel):
    id: int
    problem : str

class Question(QuestionBase):
    companies : list[CompanyQuestion]
    class Config:
        orm_mode = True